File: app/src/main/java/aws/apps/underthehood/adapters/ViewPagerAdapter.java
/*******************************************************************************
 * Copyright 2010 Alexandros Schillings
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ******************************************************************************/
package aws.apps.underthehood.adapters;

import android.util.Log;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;

public class ViewPagerAdapter extends PagerAdapter {
    private final String TAG = this.getClass().getName();

    final private ArrayList<View> mViewList;
    final private ArrayList<String> mTitleList;

    public ViewPagerAdapter() {
        mViewList = new ArrayList<>();
        mTitleList = new ArrayList<>();
    }

    public void addView(View view, String title) {
        if (view == null) {
            //Log.w(Constants.TAG, "^ ViewPagerAdapter: View is null");
            return;
        }
        mViewList.add(view);
        mTitleList.add(title.toUpperCase());
    }

    public void Clear() {
        mViewList.clear();
        mTitleList.clear();
        notifyDataSetChanged();
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object view) {
        ((ViewPager) container).removeView((View) view);
    }

    public int getCount() {
        return mViewList.size();
    }

    public Object getItem(int position) {
        return mViewList.get(position);
    }

    public long getItemId(int position) {
        return position;
    }

    public String getPageTitle(int position) {
        return mTitleList.get(position);
    }

    public View getView(int position, View convertView, ViewGroup parent) {
        return mViewList.get(position);
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        ((ViewPager) container).addView(mViewList.get(position), 0);
        return mViewList.get(position);
    }

    @Override
    public boolean isViewFromObject(View view, @NonNull Object object) {
        return view.equals(object);
    }

    public boolean shouldHideIndicator() {
        if (getCount() < 2) {
            Log.d(TAG, "^ VIEWPAGERADAPTER: There are " + getCount() + " pages. Should hide!");
            return true;
        }
        return false;
    }
}

File: app/src/main/java/aws/apps/underthehood/adapters/ViewPagerAdapter.kt
/*******************************************************************************
 * Copyright 2010 Alexandros Schillings
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package aws.apps.underthehood.adapters

import android.util.Log
import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter
import java.util.*

class ViewPagerAdapter : PagerAdapter() {
    private val TAG = this.javaClass.name
    private val views: ArrayList<View> = ArrayList()
    private val titles: ArrayList<String> = ArrayList()

    fun addView(view: View?, title: String) {
        if (view == null) {
            return
        }
        views.add(view)
        titles.add(title.toUpperCase())
    }

    fun clear() {
        views.clear()
        titles.clear()
        notifyDataSetChanged()
    }

    override fun destroyItem(container: ViewGroup, position: Int, view: Any) {
        container.removeView(view as View)
    }

    override fun getCount(): Int {
        return views.size
    }

    fun getItem(position: Int): Any {
        return views[position]
    }

    fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getPageTitle(position: Int): String? {
        return titles[position]
    }

    fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        return views[position]
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        container.addView(views[position], 0)
        return views[position]
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view == `object`
    }

    fun shouldHideIndicator(): Boolean {
        if (count < 2) {
            Log.d(TAG, "There are $count pages. Should hide!")
            return true
        }
        return false
    }

}